here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/internal/IntHandlerForEventHandlers.java

```diff
@@ -63,33 +65,107 @@ public IntHandlerForEventHandlers(final @NotNull INaksha hub) {
     if (pluginValidation instanceof ErrorResult) {
       return pluginValidation;
     }
-    return defaultStorageValidation(eventHandler);
+    return defaultHandlerValidation(eventHandler);
   }
 
-  private Result defaultStorageValidation(EventHandler eventHandler) {
-    if (isDefaultStorageHandler(eventHandler)) {
-      return storageValidationError(eventHandler);
+  private Result defaultHandlerValidation(EventHandler eventHandler) {
+    if (handlerClassMatches(DefaultStorageHandler.class, eventHandler)) {
+      return storageValidationError(eventHandler, DefaultStorageHandlerProperties.STORAGE_ID);
+    }
+    if (handlerClassMatches(DefaultViewHandler.class, eventHandler)) {
+      return propertiesValidationError(eventHandler);
+    }
+    return new SuccessResult();
+  }
+
+  private @NotNull Result propertiesValidationError(EventHandler eventHandler) {
+    Result storageValidation = storageValidationError(eventHandler, DefaultViewHandlerProperties.STORAGE_ID);
+
+    if (!(storageValidation instanceof SuccessResult)) {
+      return storageValidation;
+    }
+
+    Object spaceIdsProperty = eventHandler.getProperties().get(DefaultViewHandlerProperties.SPACE_IDS);
+    if (spaceIdsProperty == null) {
+      return new ErrorResult(
+          XyzError.ILLEGAL_ARGUMENT,
+          "Mandatory properties parameter %s missing!".formatted(DefaultViewHandlerProperties.SPACE_IDS));
+    }
+
+    if (!(spaceIdsProperty instanceof List<?>)) {
+      return new ErrorResult(
+          XyzError.ILLEGAL_ARGUMENT,
+          "Mandatory property parameter %s should be a collection of string"
+              .formatted(DefaultViewHandlerProperties.SPACE_IDS));
+    }
+    List<String> spaceIds = (List<String>) spaceIdsProperty;
+
+    if (spaceIds.isEmpty()) {
+      return new ErrorResult(
+          XyzError.ILLEGAL_ARGUMENT,
+          "Mandatory parameter %s can't be empty/blank!".formatted(DefaultViewHandlerProperties.SPACE_IDS));
+    }
+    for (String spaceId : spaceIds) {
+      if (StringUtils.isBlank(spaceId)) {
+        return new ErrorResult(
+            XyzError.ILLEGAL_ARGUMENT,
+            "Mandatory parameter %s contains space which is empty/blank!"
+                .formatted(DefaultViewHandlerProperties.SPACE_IDS));
+      }
+    }
+
+    return spaceExistenceValidation(spaceIds);
+  }
+
+  private Result spaceExistenceValidation(List<String> spaceIds) {
+
+    POp[] popQueries = new POp[spaceIds.size()];
```

Comment:
We can directly call RequestHelper.readFeaturesByIdsRequest() method, by passing the list of spaceIds. It does the same thing internally for returning ReadFeatures request back. We are doing it same way in ReadFeatureApiTask.executeFeaturesById()